tests/ModelTester.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <QAbstractItemModel>
+#include <QTest>
+#include <memory>
+
+// does some basic sanity checking (doesn't crash on invalid indexes etc.)
+class ModelTester : public QObject
```

Comment:
Not sure how this is testing anything our small city's worth of users wouldn't discover in 5 seconds, but OK ;)